### PR TITLE
[Test] API benchmarks

### PR DESCRIPTION
**Pull request by VoVAllen**
## Description
- knn
- khop
- to_block
- METIS partition
- reverse
- batch
- unbatch

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: benchmarks/benchmarks/api/bench_edge_ids.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import torch
+import numpy as np
+
+from .. import utils
+# edge_ids is not supported on cuda
+# @utils.skip_if_gpu()
+@utils.benchmark('time', timeout=1200)
+@utils.parametrize_cpu('graph_name', ['cora', 'livejournal', 'friendster'])
+@utils.parametrize_gpu('graph_name', ['cora', 'livejournal'])
+@utils.parametrize('format', ['csr'])  # csr/csc is not supported
```
Comment (jermainewang) [inline benchmarks/benchmarks/api/bench_edge_ids.py:12]: csr and coo are not supported

File: benchmarks/benchmarks/api/bench_khop.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+@utils.parametrize('format', ['coo', 'csr'])
+@utils.parametrize('k', [1, 3, 5])
+def track_time(graph_name, format, k):
+    device = utils.get_bench_device()
+    graph = utils.get_graph(graph_name, format)
+    graph = graph.to(device)
+    graph = graph.formats([format])
+    # dry run
+    dgl.khop_graph(graph, 1)
+    dgl.khop_graph(graph, 3)
```
Comment (jermainewang) [inline benchmarks/benchmarks/api/bench_khop.py:20]: `dgl.khop_graph(graph, k)`

File: benchmarks/benchmarks/api/bench_knn_graph.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+@utils.benchmark('time', timeout=60)
+@utils.parametrize('k', [3, 5, 10])
+@utils.parametrize('size', [50, 200, ])
+@utils.parametrize('dim', [16, 64, 128])
+def track_time(size, dim, k):
+    device = utils.get_bench_device()
+    features = np.random.randn(size, dim)
+    feat = torch.tensor(features, dtype=torch.float, device=device)
+    # dry run
+    dgl.knn_graph(feat, 2)
```
Comment (jermainewang) [inline benchmarks/benchmarks/api/bench_knn_graph.py:18]: `dgl.knn_graph(feat, k)`

File: benchmarks/benchmarks/api/bench_metis_partition.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+@utils.parametrize('k', [2, 4, 8])
+def track_time(graph_name, k):
+    device = utils.get_bench_device()
+    data = utils.process_data(graph_name)
+    graph = data[0]
+
+    # timing
+    t0 = time.time()
+    gg = dgl.transform.metis_partition(graph, k)
+    t1 = time.time()
```
Comment (jermainewang) [inline benchmarks/benchmarks/api/bench_metis_partition.py:21]: dry run? How slow is this?

File: benchmarks/benchmarks/api/bench_unbatch.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import time
+import dgl
+import torch
+
+from .. import utils
+
+@utils.benchmark('time')
+@utils.parametrize('batch_size', [4, 32, 256])
+def track_time(batch_size):
```
Comment (jermainewang) [inline benchmarks/benchmarks/api/bench_unbatch.py:9]: miss batching/unbatching on cuda

File: benchmarks/benchmarks/utils.py [hunk 6]
```diff
@@ -26,36 +26,70 @@ def _download(url, path, filename):
...
-        return get_livejournal()
-    else:
-        print(name + " doesn't exist")
-        return None
+# def get_graph(name):
+#     if name == 'livejournal':
+#         return get_livejournal()
+#     else:
+#         print(name + " doesn't exist")
+#         return None
```
Comment (jermainewang) [inline benchmarks/benchmarks/utils.py:92]: delete

**Review by jermainewang — CHANGES_REQUESTED** [review]

File: benchmarks/benchmarks/api/bench_unbatch.py (hunk 5 above)
Comment (VoVAllen) [reply benchmarks/benchmarks/api/bench_unbatch.py:9]: Fixed

File: benchmarks/benchmarks/api/bench_metis_partition.py (hunk 4 above)
Comment (VoVAllen) [reply benchmarks/benchmarks/api/bench_metis_partition.py:25]: Added dry run on pubmed. It takes about 3 minutes including data loading, and about 1 minute for the partition part. Changed to iterate 3 times

File: benchmarks/benchmarks/api/bench_knn_graph.py (hunk 3 above)
Comment (VoVAllen) [reply benchmarks/benchmarks/api/bench_knn_graph.py:18]: Fixed

File: benchmarks/benchmarks/api/bench_khop.py (hunk 2 above)
Comment (VoVAllen) [reply benchmarks/benchmarks/api/bench_khop.py]: Fixed

File: benchmarks/benchmarks/api/bench_edge_ids.py (hunk 1 above)
Comment (VoVAllen) [reply benchmarks/benchmarks/api/bench_edge_ids.py]: Fixed.

**Review by jermainewang — APPROVED** [review]

**Pull request merged**